src/backend/commands/typecmds.c

```diff
@@ -868,7 +868,13 @@ DefineDomain(CreateDomainStmt *stmt)
 	/* Domains never accept typmods, so no typmodin/typmodout needed */
 
 	/* Only allow typmods if GUC is set, used by babelfishpg_tsql extension */
-	if (enable_domain_typmod)
+	if (enable_domain_typmod ||
```

Comment:
It will work in a short-term, i.e., MVU for sys catalog, but I think a more desirable solution would be to modify pg_dump when considering user-defined domains, e.g.,
```
SET enable_domain_typmod = TRUE;
CREATE DOMAIN user_defined_domain_with_typmod AS user_defined_type;

SET enable_domain_typmod = FALSE;
CREATE DOMAIN user_defined_domain_without_typmod AS user_defined_type;
```

If we are able to detect whether a domain has typmod enabled during pg_dump, then we can generate the same SQL as above.

Reply:
Added above logic in pg_dump.